crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -780,13 +780,10 @@ impl ReadOnlyDevice {
 
         if let Some(mut session) = session {
             let message = session.encrypt(self, event_type, content, message_id).await?;
-
             trace!("Successfully encrypted an event");
-
             Ok((session, message))
         } else {
-            warn!("Trying to encrypt an event for a device, but no Olm session is found.",);
-
+            trace!("Trying to encrypt an event for a device, but no Olm session is found.");
```

Comment:
Re your suggested change: it looks like this information used to be attached to a `Span` in the calling method, until I messed that up in https://github.com/matrix-org/matrix-rust-sdk/pull/2862/commits/e1d2a7f17af6519b42712a9d43cbbcd378183ea2. I'd better revisit that.

> Or we could also remove it altogether, if it's already logged out in `get_missing_sessions`.

Since `get_missing_sessions` and this (called from `share_room_key`, among other places) are somewhat distinct phases of the message encryption process, it felt like it would be useful to have some way of understanding what `share_room_key` was doing if necessary. I have no string feelings though.

Reply:
> Re your suggested change: it looks like this information used to be attached to a `Span` in the calling method, until I messed that up in [e1d2a7f](https://github.com/matrix-org/matrix-rust-sdk/commit/e1d2a7f17af6519b42712a9d43cbbcd378183ea2). I'd better revisit that.

This is done in https://github.com/matrix-org/matrix-rust-sdk/pull/2873. 

For reference, the output of this logline now looks like this in EWR, when tracing is turned on.

```
matrix_sdk_crypto_wasm.js:8171 TRACE matrix_sdk_crypto::identities::device: Trying to encrypt an event for a device, but no Olm session is found.
    at /home/rav/work/matrix-rust-sdk/crates/matrix-sdk-crypto/src/identities/device.rs:776
    in matrix_sdk_crypto::identities::device::encrypt with recipient="<user_id>" recipient_device="APJEQVRJIP" recipient_key=Some("curve25519:63272b+OOSZT7NrBf1EltxSp8vy9I1GW4z4wVVekKDA") message_id="cebccab964f742e0b685f8bbe54326a9"